realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1093,29 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Returns a distinct set of objects of a specific class. As a Realm is unordered, it is
+     * undefined which objects are returned in case of multiple occurrences.
```

Comment:
@beeender I don't think so. I get the sense from your comment that Java side classes should ensemble their counterpart in core side. i.e. `void distinct()` -> `void distinct()`. Not only this makes my job easier but also we'd have one unified idea around core and whoever will follow would have short amount of time to get under the skin. 

I'll dearly follow your pointers around javadoc and coding style.